subsys/net/ip/icmpv4.c

```diff
@@ -599,6 +599,17 @@ int net_icmpv4_send_error(struct net_pkt *orig, uint8_t type, uint8_t code)
 		}
 	}
 
+	if (net_ipv4_is_addr_bcast(net_pkt_iface(orig),
+				   (struct in_addr *)ip_hdr->dst)) {
```

Comment:
This line result in compilation error, need to change `(struct in_addr *)ip_hdr->dst` → `&ip_hdr->dst`